mitosheet/mitosheet/mito_dash/v1/spreadsheet.py

```diff
@@ -149,74 +201,78 @@ def get_result(self):
     """
-    Provides a server-side callback relating the values of one or more `Output` items 
-    to a single spreadsheet with the given id.
+    Returns a list of all the Input components that are Spreadsheet components, and their indexes
 
-    Example usage:
-    ```
-    from mitosheet.mito_dash.v1 import Spreadsheet, spreadsheet_callback
-    from dash import Dash, html, Output
+    TODO: maybe need to handle lists of Outputs and Inputs and States
+    """
+    result = []
+    callback_index = 0
+    for index, arg in enumerate(args):
 
-    import pandas as pd
 
-    df = pd.DataFrame({'A': [1, 2, 3]})
+        if (isinstance(arg, Input) or isinstance(arg, State)) and arg.component_id is not None:
+            component_id = arg.component_id
+            component_property = arg.component_property
 
-    app = Dash(__name__)
+            # Mito currently supports the following properities:
+            # - mito_return_value
 
-    app.layout = html.Div([
-        Spreadsheet(df, id='mito-dash-wrapper'),
-        html.Div(id='output'),    
-    ])
+            if component_property == 'mito_return_value':
+                result.append((component_id, index, callback_index))
 
-    @spreadsheet_callback(
-        Output('output', 'children'),
-        input_id='mito-dash-wrapper',
-    )
-    def update_output(spreadsheet_result):
-        dfs = spreadsheet_result.dfs()
-        return f'Output: {str(dfs)}'
+            # If they try and access other properties of the Spreadsheet component, we raise an error
+            if component_property in ['all_json', ]:
+                raise Exception(f"Cannot access property {component_property} of Spreadsheet component with id {component_id}. Please only access the mito_return_value property.")
 
-    
-    if __name__ == '__main__':
-        app.run_server(debug=True)
-    ```
+        if (isinstance(arg, Input) or isinstance(arg, State)):
+            callback_index += 1
+            
+    return result
 
-    """
-    if input_id is None:
-        raise ValueError('input_id must be provided')
-    
-    # Find the Spreadsheet component with the given id - searching all objects in this Python runtime
-    callback_component = None
-    components = [
-        obj for obj in gc.get_objects()
-        if isinstance(obj, Spreadsheet) and getattr(obj, 'mito_id', None) == input_id
-    ]
 
-    if len(components) > 0:
-        callback_component = components[0]
+def mito_callback(*args, **kwargs):
+    # First, check if there are any args that are Inputs that contain a mito_id
+    indexes = get_spreadsheet_id_and_indexes_in_callback_args(*args)
 
-    if callback_component is None:
-        raise ValueError(f'Could not find Spreadsheet with id {input_id}. Make sure the input_id matches the id of a spreadsheet component')
+    # If there are no spreadsheet input components, then we just call the regular callback
+    if len(indexes) == 0:
```

Comment:
If there is no spreadsheet should we even be supporting this case? Should we just through an error instead? 

Maybe the most common reason someone is going to be here is just because they don't understand how to create a mitosheet and the error message could be useful.

Reply:
They might just be iteratively developing, and not currently have a Mitosheet in their callback. This is probably pretty frequent!